{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Cell Types Project\n",
    "*Replace this with the name of your project*"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Team Member Names & Contributions\n",
    "Group 3\n",
    "- **Chloe Nguyen (A16794414)**: Responsible for data wrangling, data visualization\n",
    "- **Amber Selstad (A16898643)**: "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Abstract"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "*Fill in your 3-4 sentence abstract here*"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Research Question"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "*Fill in your research question here*"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Background and Prior Work"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "*Fill in your background and prior work here (~500 words). You are welcome to use additional subheadings. You should also include a paragraph describing each dataset and how you'll be using them.* \n",
    "\n",
    "### References (include links):\n",
    "(1) \n",
    "\n",
    "(2)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Hypothesis\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "*Fill in your hypotheses here*"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Setup"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "## This code imports packages essential for downstream data wrangling, analysis, manipulation, etc. Figures are plotted within the Jupyter notebook\n",
    "import numpy as np\n",
    "import pandas as pd\n",
    "import matplotlib as mpl\n",
    "import matplotlib.pyplot as plt\n",
    "%matplotlib inline\n",
    "from scipy import stats"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Data Wrangling"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Describe your data wrangling steps here."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Both datasets of interest for our project come from the Allen Brain Institute, specifically the Allen Cell Types website (https://celltypes.brain-map.org/data) and the Allen RNA-seq website (https://celltypes.brain-map.org/data). The Cell Types website provides an extensive database of electrophysiological and morphogical features of both human and mouse cells, which we will utilize to filter for the samples needed to answer our research question: how do the gene expression and electrophysiological profiles of human neurons differ by whether they are spiny or aspiny?\n",
    "\n",
    "The Allen Software Development Kit (AllenSDK) is a core package that facilitates reading and processing Allen Brain Institute data like the Cell Types one. First, once AllenSDK is installed, the 'CellTypesCache' module can extract and access the Cell Types database within a Python environment, while the 'CellTypesApi' module then enables us to query the database. After initalizing the cache to ctc, we can then begin filtering this data."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>reporter_status</th>\n",
       "      <th>cell_soma_location</th>\n",
       "      <th>species</th>\n",
       "      <th>name</th>\n",
       "      <th>structure_layer_name</th>\n",
       "      <th>structure_area_id</th>\n",
       "      <th>structure_area_abbrev</th>\n",
       "      <th>transgenic_line</th>\n",
       "      <th>dendrite_type</th>\n",
       "      <th>apical</th>\n",
       "      <th>reconstruction_type</th>\n",
       "      <th>disease_state</th>\n",
       "      <th>donor_id</th>\n",
       "      <th>structure_hemisphere</th>\n",
       "      <th>normalized_depth</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>id</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>525011903</th>\n",
       "      <td>None</td>\n",
       "      <td>[273.0, 354.0, 216.0]</td>\n",
       "      <td>Homo Sapiens</td>\n",
       "      <td>H16.03.003.01.14.02</td>\n",
       "      <td>3</td>\n",
       "      <td>12113</td>\n",
       "      <td>FroL</td>\n",
       "      <td></td>\n",
       "      <td>spiny</td>\n",
       "      <td>intact</td>\n",
       "      <td>None</td>\n",
       "      <td>epilepsy</td>\n",
       "      <td>524848408</td>\n",
       "      <td>right</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>528642047</th>\n",
       "      <td>None</td>\n",
       "      <td>[69.0, 254.0, 96.0]</td>\n",
       "      <td>Homo Sapiens</td>\n",
       "      <td>H16.06.009.01.02.06.05</td>\n",
       "      <td>5</td>\n",
       "      <td>12141</td>\n",
       "      <td>MTG</td>\n",
       "      <td></td>\n",
       "      <td>aspiny</td>\n",
       "      <td>NA</td>\n",
       "      <td>None</td>\n",
       "      <td>epilepsy</td>\n",
       "      <td>528574320</td>\n",
       "      <td>left</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>537256313</th>\n",
       "      <td>None</td>\n",
       "      <td>[322.0, 255.0, 92.0]</td>\n",
       "      <td>Homo Sapiens</td>\n",
       "      <td>H16.03.006.01.05.02</td>\n",
       "      <td>4</td>\n",
       "      <td>12141</td>\n",
       "      <td>MTG</td>\n",
       "      <td></td>\n",
       "      <td>spiny</td>\n",
       "      <td>truncated</td>\n",
       "      <td>None</td>\n",
       "      <td>epilepsy</td>\n",
       "      <td>536912860</td>\n",
       "      <td>right</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>519832676</th>\n",
       "      <td>None</td>\n",
       "      <td>[79.0, 273.0, 91.0]</td>\n",
       "      <td>Homo Sapiens</td>\n",
       "      <td>H16.03.001.01.09.01</td>\n",
       "      <td>3</td>\n",
       "      <td>12141</td>\n",
       "      <td>MTG</td>\n",
       "      <td></td>\n",
       "      <td>spiny</td>\n",
       "      <td>truncated</td>\n",
       "      <td>full</td>\n",
       "      <td>epilepsy</td>\n",
       "      <td>518641172</td>\n",
       "      <td>left</td>\n",
       "      <td>0.290951</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>596020931</th>\n",
       "      <td>None</td>\n",
       "      <td>[66.0, 220.0, 105.0]</td>\n",
       "      <td>Homo Sapiens</td>\n",
       "      <td>H17.06.009.11.04.02</td>\n",
       "      <td>4</td>\n",
       "      <td>12141</td>\n",
       "      <td>MTG</td>\n",
       "      <td></td>\n",
       "      <td>aspiny</td>\n",
       "      <td>NA</td>\n",
       "      <td>full</td>\n",
       "      <td>tumor</td>\n",
       "      <td>595954915</td>\n",
       "      <td>left</td>\n",
       "      <td>0.497825</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "          reporter_status     cell_soma_location       species  \\\n",
       "id                                                               \n",
       "525011903            None  [273.0, 354.0, 216.0]  Homo Sapiens   \n",
       "528642047            None    [69.0, 254.0, 96.0]  Homo Sapiens   \n",
       "537256313            None   [322.0, 255.0, 92.0]  Homo Sapiens   \n",
       "519832676            None    [79.0, 273.0, 91.0]  Homo Sapiens   \n",
       "596020931            None   [66.0, 220.0, 105.0]  Homo Sapiens   \n",
       "\n",
       "                             name structure_layer_name  structure_area_id  \\\n",
       "id                                                                          \n",
       "525011903     H16.03.003.01.14.02                    3              12113   \n",
       "528642047  H16.06.009.01.02.06.05                    5              12141   \n",
       "537256313     H16.03.006.01.05.02                    4              12141   \n",
       "519832676     H16.03.001.01.09.01                    3              12141   \n",
       "596020931     H17.06.009.11.04.02                    4              12141   \n",
       "\n",
       "          structure_area_abbrev transgenic_line dendrite_type     apical  \\\n",
       "id                                                                         \n",
       "525011903                  FroL                         spiny     intact   \n",
       "528642047                   MTG                        aspiny         NA   \n",
       "537256313                   MTG                         spiny  truncated   \n",
       "519832676                   MTG                         spiny  truncated   \n",
       "596020931                   MTG                        aspiny         NA   \n",
       "\n",
       "          reconstruction_type disease_state   donor_id structure_hemisphere  \\\n",
       "id                                                                            \n",
       "525011903                None      epilepsy  524848408                right   \n",
       "528642047                None      epilepsy  528574320                 left   \n",
       "537256313                None      epilepsy  536912860                right   \n",
       "519832676                full      epilepsy  518641172                 left   \n",
       "596020931                full         tumor  595954915                 left   \n",
       "\n",
       "           normalized_depth  \n",
       "id                           \n",
       "525011903               NaN  \n",
       "528642047               NaN  \n",
       "537256313               NaN  \n",
       "519832676          0.290951  \n",
       "596020931          0.497825  "
      ]
     },
     "execution_count": 2,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Importing the Allen Cell Types database into Python\n",
    "from allensdk.core.cell_types_cache import CellTypesCache\n",
    "from allensdk.api.queries.cell_types_api import CellTypesApi\n",
    "ctc = CellTypesCache(manifest_file='cell_types/manifest.json')\n",
    "\n",
    "# Creating a dataframe with only human cells\n",
    "human_cells = ctc.get_cells(species=[CellTypesApi.HUMAN])\n",
    "human_df = pd.DataFrame(human_cells)\n",
    "\n",
    "# Setting the index to the specimen id instead of arbitrary index values\n",
    "human_df = human_df.set_index('id') \n",
    "human_df.head(5) # Showcasing the first 5 rows of this dataframe"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "For the scope of our project, we are interested in comparing spiny and aspiny neurons across all provided brain regions in the Cell Types Database instead a select few lobes or regions. This research method can provide a broader generalized overview of spiny versus aspiny neuron function within the human cerebral cortex."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Creating separate datasets for spiny versus aspiny neurons within all brain regions\n",
    "human_spiny = human_df[human_df['dendrite_type'] == 'spiny']\n",
    "human_aspiny = human_df[human_df['dendrite_type'] == 'aspiny']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>reporter_status</th>\n",
       "      <th>cell_soma_location</th>\n",
       "      <th>species</th>\n",
       "      <th>name</th>\n",
       "      <th>structure_layer_name</th>\n",
       "      <th>structure_area_id</th>\n",
       "      <th>structure_area_abbrev</th>\n",
       "      <th>transgenic_line</th>\n",
       "      <th>dendrite_type</th>\n",
       "      <th>apical</th>\n",
       "      <th>reconstruction_type</th>\n",
       "      <th>disease_state</th>\n",
       "      <th>donor_id</th>\n",
       "      <th>structure_hemisphere</th>\n",
       "      <th>normalized_depth</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>id</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>525011903</th>\n",
       "      <td>None</td>\n",
       "      <td>[273.0, 354.0, 216.0]</td>\n",
       "      <td>Homo Sapiens</td>\n",
       "      <td>H16.03.003.01.14.02</td>\n",
       "      <td>3</td>\n",
       "      <td>12113</td>\n",
       "      <td>FroL</td>\n",
       "      <td></td>\n",
       "      <td>spiny</td>\n",
       "      <td>intact</td>\n",
       "      <td>None</td>\n",
       "      <td>epilepsy</td>\n",
       "      <td>524848408</td>\n",
       "      <td>right</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>537256313</th>\n",
       "      <td>None</td>\n",
       "      <td>[322.0, 255.0, 92.0]</td>\n",
       "      <td>Homo Sapiens</td>\n",
       "      <td>H16.03.006.01.05.02</td>\n",
       "      <td>4</td>\n",
       "      <td>12141</td>\n",
       "      <td>MTG</td>\n",
       "      <td></td>\n",
       "      <td>spiny</td>\n",
       "      <td>truncated</td>\n",
       "      <td>None</td>\n",
       "      <td>epilepsy</td>\n",
       "      <td>536912860</td>\n",
       "      <td>right</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>519832676</th>\n",
       "      <td>None</td>\n",
       "      <td>[79.0, 273.0, 91.0]</td>\n",
       "      <td>Homo Sapiens</td>\n",
       "      <td>H16.03.001.01.09.01</td>\n",
       "      <td>3</td>\n",
       "      <td>12141</td>\n",
       "      <td>MTG</td>\n",
       "      <td></td>\n",
       "      <td>spiny</td>\n",
       "      <td>truncated</td>\n",
       "      <td>full</td>\n",
       "      <td>epilepsy</td>\n",
       "      <td>518641172</td>\n",
       "      <td>left</td>\n",
       "      <td>0.290951</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>569095789</th>\n",
       "      <td>None</td>\n",
       "      <td>[110.0, 122.0, 240.0]</td>\n",
       "      <td>Homo Sapiens</td>\n",
       "      <td>H17.06.004.11.05.04</td>\n",
       "      <td>2</td>\n",
       "      <td>12136</td>\n",
       "      <td>AnG</td>\n",
       "      <td></td>\n",
       "      <td>spiny</td>\n",
       "      <td>intact</td>\n",
       "      <td>full</td>\n",
       "      <td>tumor</td>\n",
       "      <td>569008241</td>\n",
       "      <td>left</td>\n",
       "      <td>0.056400</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>545608578</th>\n",
       "      <td>None</td>\n",
       "      <td>[312.0, 280.0, 89.0]</td>\n",
       "      <td>Homo Sapiens</td>\n",
       "      <td>H16.03.010.13.06.01</td>\n",
       "      <td>3</td>\n",
       "      <td>12141</td>\n",
       "      <td>MTG</td>\n",
       "      <td></td>\n",
       "      <td>spiny</td>\n",
       "      <td>intact</td>\n",
       "      <td>None</td>\n",
       "      <td>epilepsy</td>\n",
       "      <td>545510854</td>\n",
       "      <td>right</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "          reporter_status     cell_soma_location       species  \\\n",
       "id                                                               \n",
       "525011903            None  [273.0, 354.0, 216.0]  Homo Sapiens   \n",
       "537256313            None   [322.0, 255.0, 92.0]  Homo Sapiens   \n",
       "519832676            None    [79.0, 273.0, 91.0]  Homo Sapiens   \n",
       "569095789            None  [110.0, 122.0, 240.0]  Homo Sapiens   \n",
       "545608578            None   [312.0, 280.0, 89.0]  Homo Sapiens   \n",
       "\n",
       "                          name structure_layer_name  structure_area_id  \\\n",
       "id                                                                       \n",
       "525011903  H16.03.003.01.14.02                    3              12113   \n",
       "537256313  H16.03.006.01.05.02                    4              12141   \n",
       "519832676  H16.03.001.01.09.01                    3              12141   \n",
       "569095789  H17.06.004.11.05.04                    2              12136   \n",
       "545608578  H16.03.010.13.06.01                    3              12141   \n",
       "\n",
       "          structure_area_abbrev transgenic_line dendrite_type     apical  \\\n",
       "id                                                                         \n",
       "525011903                  FroL                         spiny     intact   \n",
       "537256313                   MTG                         spiny  truncated   \n",
       "519832676                   MTG                         spiny  truncated   \n",
       "569095789                   AnG                         spiny     intact   \n",
       "545608578                   MTG                         spiny     intact   \n",
       "\n",
       "          reconstruction_type disease_state   donor_id structure_hemisphere  \\\n",
       "id                                                                            \n",
       "525011903                None      epilepsy  524848408                right   \n",
       "537256313                None      epilepsy  536912860                right   \n",
       "519832676                full      epilepsy  518641172                 left   \n",
       "569095789                full         tumor  569008241                 left   \n",
       "545608578                None      epilepsy  545510854                right   \n",
       "\n",
       "           normalized_depth  \n",
       "id                           \n",
       "525011903               NaN  \n",
       "537256313               NaN  \n",
       "519832676          0.290951  \n",
       "569095789          0.056400  \n",
       "545608578               NaN  "
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Checking the dataframe to ensure it has the correct rows\n",
    "human_spiny.head(5) "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>reporter_status</th>\n",
       "      <th>cell_soma_location</th>\n",
       "      <th>species</th>\n",
       "      <th>name</th>\n",
       "      <th>structure_layer_name</th>\n",
       "      <th>structure_area_id</th>\n",
       "      <th>structure_area_abbrev</th>\n",
       "      <th>transgenic_line</th>\n",
       "      <th>dendrite_type</th>\n",
       "      <th>apical</th>\n",
       "      <th>reconstruction_type</th>\n",
       "      <th>disease_state</th>\n",
       "      <th>donor_id</th>\n",
       "      <th>structure_hemisphere</th>\n",
       "      <th>normalized_depth</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>id</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>528642047</th>\n",
       "      <td>None</td>\n",
       "      <td>[69.0, 254.0, 96.0]</td>\n",
       "      <td>Homo Sapiens</td>\n",
       "      <td>H16.06.009.01.02.06.05</td>\n",
       "      <td>5</td>\n",
       "      <td>12141</td>\n",
       "      <td>MTG</td>\n",
       "      <td></td>\n",
       "      <td>aspiny</td>\n",
       "      <td>NA</td>\n",
       "      <td>None</td>\n",
       "      <td>epilepsy</td>\n",
       "      <td>528574320</td>\n",
       "      <td>left</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>596020931</th>\n",
       "      <td>None</td>\n",
       "      <td>[66.0, 220.0, 105.0]</td>\n",
       "      <td>Homo Sapiens</td>\n",
       "      <td>H17.06.009.11.04.02</td>\n",
       "      <td>4</td>\n",
       "      <td>12141</td>\n",
       "      <td>MTG</td>\n",
       "      <td></td>\n",
       "      <td>aspiny</td>\n",
       "      <td>NA</td>\n",
       "      <td>full</td>\n",
       "      <td>tumor</td>\n",
       "      <td>595954915</td>\n",
       "      <td>left</td>\n",
       "      <td>0.497825</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>576095991</th>\n",
       "      <td>None</td>\n",
       "      <td>[274.0, 331.0, 228.0]</td>\n",
       "      <td>Homo Sapiens</td>\n",
       "      <td>H17.06.007.11.08.01</td>\n",
       "      <td>5</td>\n",
       "      <td>12116</td>\n",
       "      <td>MFG</td>\n",
       "      <td></td>\n",
       "      <td>aspiny</td>\n",
       "      <td>NA</td>\n",
       "      <td>full</td>\n",
       "      <td>tumor</td>\n",
       "      <td>576060516</td>\n",
       "      <td>right</td>\n",
       "      <td>0.673428</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>531520637</th>\n",
       "      <td>None</td>\n",
       "      <td>[127.0, 354.0, 219.0]</td>\n",
       "      <td>Homo Sapiens</td>\n",
       "      <td>H16.06.007.01.05.03</td>\n",
       "      <td>3</td>\n",
       "      <td>12113</td>\n",
       "      <td>FroL</td>\n",
       "      <td></td>\n",
       "      <td>aspiny</td>\n",
       "      <td>NA</td>\n",
       "      <td>full</td>\n",
       "      <td>tumor</td>\n",
       "      <td>518229880</td>\n",
       "      <td>left</td>\n",
       "      <td>0.388676</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>596049209</th>\n",
       "      <td>None</td>\n",
       "      <td>[66.0, 220.0, 105.0]</td>\n",
       "      <td>Homo Sapiens</td>\n",
       "      <td>H17.06.009.11.04.05</td>\n",
       "      <td>1</td>\n",
       "      <td>12141</td>\n",
       "      <td>MTG</td>\n",
       "      <td></td>\n",
       "      <td>aspiny</td>\n",
       "      <td>NA</td>\n",
       "      <td>None</td>\n",
       "      <td>tumor</td>\n",
       "      <td>595954915</td>\n",
       "      <td>left</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "          reporter_status     cell_soma_location       species  \\\n",
       "id                                                               \n",
       "528642047            None    [69.0, 254.0, 96.0]  Homo Sapiens   \n",
       "596020931            None   [66.0, 220.0, 105.0]  Homo Sapiens   \n",
       "576095991            None  [274.0, 331.0, 228.0]  Homo Sapiens   \n",
       "531520637            None  [127.0, 354.0, 219.0]  Homo Sapiens   \n",
       "596049209            None   [66.0, 220.0, 105.0]  Homo Sapiens   \n",
       "\n",
       "                             name structure_layer_name  structure_area_id  \\\n",
       "id                                                                          \n",
       "528642047  H16.06.009.01.02.06.05                    5              12141   \n",
       "596020931     H17.06.009.11.04.02                    4              12141   \n",
       "576095991     H17.06.007.11.08.01                    5              12116   \n",
       "531520637     H16.06.007.01.05.03                    3              12113   \n",
       "596049209     H17.06.009.11.04.05                    1              12141   \n",
       "\n",
       "          structure_area_abbrev transgenic_line dendrite_type apical  \\\n",
       "id                                                                     \n",
       "528642047                   MTG                        aspiny     NA   \n",
       "596020931                   MTG                        aspiny     NA   \n",
       "576095991                   MFG                        aspiny     NA   \n",
       "531520637                  FroL                        aspiny     NA   \n",
       "596049209                   MTG                        aspiny     NA   \n",
       "\n",
       "          reconstruction_type disease_state   donor_id structure_hemisphere  \\\n",
       "id                                                                            \n",
       "528642047                None      epilepsy  528574320                 left   \n",
       "596020931                full         tumor  595954915                 left   \n",
       "576095991                full         tumor  576060516                right   \n",
       "531520637                full         tumor  518229880                 left   \n",
       "596049209                None         tumor  595954915                 left   \n",
       "\n",
       "           normalized_depth  \n",
       "id                           \n",
       "528642047               NaN  \n",
       "596020931          0.497825  \n",
       "576095991          0.673428  \n",
       "531520637          0.388676  \n",
       "596049209               NaN  "
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Checking the dataframe to ensure it has the correct rows\n",
    "human_aspiny.head(5)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "After creating separate dataframes for our spiny versus aspiny cell data, we then need to import their corresponding electrophysiological data. This enables us to compare and contrast their behaviors given their different morphological features, especially as spiny and aspiny neurons are known to perform different functions and responses to inputs/stimuli. This can be done with the get_ephys_data function provided by the AllenSDK package."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Index(['adaptation', 'avg_isi', 'electrode_0_pa', 'f_i_curve_slope',\n",
      "       'fast_trough_t_long_square', 'fast_trough_t_ramp',\n",
      "       'fast_trough_t_short_square', 'fast_trough_v_long_square',\n",
      "       'fast_trough_v_ramp', 'fast_trough_v_short_square', 'has_burst',\n",
      "       'has_delay', 'has_pause', 'id', 'input_resistance_mohm', 'latency',\n",
      "       'peak_t_long_square', 'peak_t_ramp', 'peak_t_short_square',\n",
      "       'peak_v_long_square', 'peak_v_ramp', 'peak_v_short_square',\n",
      "       'rheobase_sweep_id', 'rheobase_sweep_number', 'ri', 'sag', 'seal_gohm',\n",
      "       'slow_trough_t_long_square', 'slow_trough_t_ramp',\n",
      "       'slow_trough_t_short_square', 'slow_trough_v_long_square',\n",
      "       'slow_trough_v_ramp', 'slow_trough_v_short_square', 'tau',\n",
      "       'threshold_i_long_square', 'threshold_i_ramp',\n",
      "       'threshold_i_short_square', 'threshold_t_long_square',\n",
      "       'threshold_t_ramp', 'threshold_t_short_square',\n",
      "       'threshold_v_long_square', 'threshold_v_ramp',\n",
      "       'threshold_v_short_square', 'thumbnail_sweep_id',\n",
      "       'trough_t_long_square', 'trough_t_ramp', 'trough_t_short_square',\n",
      "       'trough_v_long_square', 'trough_v_ramp', 'trough_v_short_square',\n",
      "       'upstroke_downstroke_ratio_long_square',\n",
      "       'upstroke_downstroke_ratio_ramp',\n",
      "       'upstroke_downstroke_ratio_short_square', 'vm_for_sag', 'vrest'],\n",
      "      dtype='object')\n"
     ]
    }
   ],
   "source": [
    "ephys_features = ctc.get_ephys_features()\n",
    "ephys_features = pd.DataFrame(ephys_features)\n",
    "ephys_features = ephys_features.set_index('specimen_id') # Setting the index by specimen_id, similar to how the previous dataframes were set to 'id' \n",
    "print(ephys_features.columns)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "We then have to merge together our spiny and aspiny dataframes with their respective ephys data, doing so with the pandas 'join' with the default 'inner' function that will only merge entrices that appear in both dataframes."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [],
   "source": [
    "human_spiny_ephys = human_spiny.join(ephys_features, how='inner')\n",
    "human_aspiny_ephys = human_aspiny.join(ephys_features, how='inner')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>reporter_status</th>\n",
       "      <th>cell_soma_location</th>\n",
       "      <th>species</th>\n",
       "      <th>name</th>\n",
       "      <th>structure_layer_name</th>\n",
       "      <th>structure_area_id</th>\n",
       "      <th>structure_area_abbrev</th>\n",
       "      <th>transgenic_line</th>\n",
       "      <th>dendrite_type</th>\n",
       "      <th>apical</th>\n",
       "      <th>...</th>\n",
       "      <th>trough_t_ramp</th>\n",
       "      <th>trough_t_short_square</th>\n",
       "      <th>trough_v_long_square</th>\n",
       "      <th>trough_v_ramp</th>\n",
       "      <th>trough_v_short_square</th>\n",
       "      <th>upstroke_downstroke_ratio_long_square</th>\n",
       "      <th>upstroke_downstroke_ratio_ramp</th>\n",
       "      <th>upstroke_downstroke_ratio_short_square</th>\n",
       "      <th>vm_for_sag</th>\n",
       "      <th>vrest</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>525011903</th>\n",
       "      <td>None</td>\n",
       "      <td>[273.0, 354.0, 216.0]</td>\n",
       "      <td>Homo Sapiens</td>\n",
       "      <td>H16.03.003.01.14.02</td>\n",
       "      <td>3</td>\n",
       "      <td>12113</td>\n",
       "      <td>FroL</td>\n",
       "      <td></td>\n",
       "      <td>spiny</td>\n",
       "      <td>intact</td>\n",
       "      <td>...</td>\n",
       "      <td>4.134987</td>\n",
       "      <td>1.375253</td>\n",
       "      <td>-53.968754</td>\n",
       "      <td>-59.510420</td>\n",
       "      <td>-71.197919</td>\n",
       "      <td>2.895461</td>\n",
       "      <td>2.559876</td>\n",
       "      <td>3.099787</td>\n",
       "      <td>-88.843758</td>\n",
       "      <td>-70.561035</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>537256313</th>\n",
       "      <td>None</td>\n",
       "      <td>[322.0, 255.0, 92.0]</td>\n",
       "      <td>Homo Sapiens</td>\n",
       "      <td>H16.03.006.01.05.02</td>\n",
       "      <td>4</td>\n",
       "      <td>12141</td>\n",
       "      <td>MTG</td>\n",
       "      <td></td>\n",
       "      <td>spiny</td>\n",
       "      <td>truncated</td>\n",
       "      <td>...</td>\n",
       "      <td>5.694547</td>\n",
       "      <td>1.389900</td>\n",
       "      <td>-52.125004</td>\n",
       "      <td>-51.520836</td>\n",
       "      <td>-72.900002</td>\n",
       "      <td>3.121182</td>\n",
       "      <td>3.464528</td>\n",
       "      <td>3.054681</td>\n",
       "      <td>-87.531250</td>\n",
       "      <td>-72.628105</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>519832676</th>\n",
       "      <td>None</td>\n",
       "      <td>[79.0, 273.0, 91.0]</td>\n",
       "      <td>Homo Sapiens</td>\n",
       "      <td>H16.03.001.01.09.01</td>\n",
       "      <td>3</td>\n",
       "      <td>12141</td>\n",
       "      <td>MTG</td>\n",
       "      <td></td>\n",
       "      <td>spiny</td>\n",
       "      <td>truncated</td>\n",
       "      <td>...</td>\n",
       "      <td>9.962780</td>\n",
       "      <td>1.211020</td>\n",
       "      <td>-53.875004</td>\n",
       "      <td>-52.416668</td>\n",
       "      <td>-73.693753</td>\n",
       "      <td>4.574865</td>\n",
       "      <td>3.817988</td>\n",
       "      <td>4.980603</td>\n",
       "      <td>-84.218758</td>\n",
       "      <td>-72.547661</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>569095789</th>\n",
       "      <td>None</td>\n",
       "      <td>[110.0, 122.0, 240.0]</td>\n",
       "      <td>Homo Sapiens</td>\n",
       "      <td>H17.06.004.11.05.04</td>\n",
       "      <td>2</td>\n",
       "      <td>12136</td>\n",
       "      <td>AnG</td>\n",
       "      <td></td>\n",
       "      <td>spiny</td>\n",
       "      <td>intact</td>\n",
       "      <td>...</td>\n",
       "      <td>5.483953</td>\n",
       "      <td>1.225900</td>\n",
       "      <td>-51.406254</td>\n",
       "      <td>-59.552087</td>\n",
       "      <td>-81.875008</td>\n",
       "      <td>2.199872</td>\n",
       "      <td>2.220031</td>\n",
       "      <td>2.235442</td>\n",
       "      <td>-95.843758</td>\n",
       "      <td>-81.065971</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>545608578</th>\n",
       "      <td>None</td>\n",
       "      <td>[312.0, 280.0, 89.0]</td>\n",
       "      <td>Homo Sapiens</td>\n",
       "      <td>H16.03.010.13.06.01</td>\n",
       "      <td>3</td>\n",
       "      <td>12141</td>\n",
       "      <td>MTG</td>\n",
       "      <td></td>\n",
       "      <td>spiny</td>\n",
       "      <td>intact</td>\n",
       "      <td>...</td>\n",
       "      <td>22.069340</td>\n",
       "      <td>1.112633</td>\n",
       "      <td>-54.343754</td>\n",
       "      <td>-54.968751</td>\n",
       "      <td>-75.156258</td>\n",
       "      <td>3.675430</td>\n",
       "      <td>3.665890</td>\n",
       "      <td>3.586321</td>\n",
       "      <td>-78.500000</td>\n",
       "      <td>-74.496262</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 70 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "          reporter_status     cell_soma_location       species  \\\n",
       "525011903            None  [273.0, 354.0, 216.0]  Homo Sapiens   \n",
       "537256313            None   [322.0, 255.0, 92.0]  Homo Sapiens   \n",
       "519832676            None    [79.0, 273.0, 91.0]  Homo Sapiens   \n",
       "569095789            None  [110.0, 122.0, 240.0]  Homo Sapiens   \n",
       "545608578            None   [312.0, 280.0, 89.0]  Homo Sapiens   \n",
       "\n",
       "                          name structure_layer_name  structure_area_id  \\\n",
       "525011903  H16.03.003.01.14.02                    3              12113   \n",
       "537256313  H16.03.006.01.05.02                    4              12141   \n",
       "519832676  H16.03.001.01.09.01                    3              12141   \n",
       "569095789  H17.06.004.11.05.04                    2              12136   \n",
       "545608578  H16.03.010.13.06.01                    3              12141   \n",
       "\n",
       "          structure_area_abbrev transgenic_line dendrite_type     apical  ...  \\\n",
       "525011903                  FroL                         spiny     intact  ...   \n",
       "537256313                   MTG                         spiny  truncated  ...   \n",
       "519832676                   MTG                         spiny  truncated  ...   \n",
       "569095789                   AnG                         spiny     intact  ...   \n",
       "545608578                   MTG                         spiny     intact  ...   \n",
       "\n",
       "          trough_t_ramp trough_t_short_square  trough_v_long_square  \\\n",
       "525011903      4.134987              1.375253            -53.968754   \n",
       "537256313      5.694547              1.389900            -52.125004   \n",
       "519832676      9.962780              1.211020            -53.875004   \n",
       "569095789      5.483953              1.225900            -51.406254   \n",
       "545608578     22.069340              1.112633            -54.343754   \n",
       "\n",
       "          trough_v_ramp  trough_v_short_square  \\\n",
       "525011903    -59.510420             -71.197919   \n",
       "537256313    -51.520836             -72.900002   \n",
       "519832676    -52.416668             -73.693753   \n",
       "569095789    -59.552087             -81.875008   \n",
       "545608578    -54.968751             -75.156258   \n",
       "\n",
       "           upstroke_downstroke_ratio_long_square  \\\n",
       "525011903                               2.895461   \n",
       "537256313                               3.121182   \n",
       "519832676                               4.574865   \n",
       "569095789                               2.199872   \n",
       "545608578                               3.675430   \n",
       "\n",
       "           upstroke_downstroke_ratio_ramp  \\\n",
       "525011903                        2.559876   \n",
       "537256313                        3.464528   \n",
       "519832676                        3.817988   \n",
       "569095789                        2.220031   \n",
       "545608578                        3.665890   \n",
       "\n",
       "           upstroke_downstroke_ratio_short_square  vm_for_sag      vrest  \n",
       "525011903                                3.099787  -88.843758 -70.561035  \n",
       "537256313                                3.054681  -87.531250 -72.628105  \n",
       "519832676                                4.980603  -84.218758 -72.547661  \n",
       "569095789                                2.235442  -95.843758 -81.065971  \n",
       "545608578                                3.586321  -78.500000 -74.496262  \n",
       "\n",
       "[5 rows x 70 columns]"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "human_spiny_ephys.head(5) # Checking the dataframe to ensure the merge was successful"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>reporter_status</th>\n",
       "      <th>cell_soma_location</th>\n",
       "      <th>species</th>\n",
       "      <th>name</th>\n",
       "      <th>structure_layer_name</th>\n",
       "      <th>structure_area_id</th>\n",
       "      <th>structure_area_abbrev</th>\n",
       "      <th>transgenic_line</th>\n",
       "      <th>dendrite_type</th>\n",
       "      <th>apical</th>\n",
       "      <th>...</th>\n",
       "      <th>trough_t_ramp</th>\n",
       "      <th>trough_t_short_square</th>\n",
       "      <th>trough_v_long_square</th>\n",
       "      <th>trough_v_ramp</th>\n",
       "      <th>trough_v_short_square</th>\n",
       "      <th>upstroke_downstroke_ratio_long_square</th>\n",
       "      <th>upstroke_downstroke_ratio_ramp</th>\n",
       "      <th>upstroke_downstroke_ratio_short_square</th>\n",
       "      <th>vm_for_sag</th>\n",
       "      <th>vrest</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>528642047</th>\n",
       "      <td>None</td>\n",
       "      <td>[69.0, 254.0, 96.0]</td>\n",
       "      <td>Homo Sapiens</td>\n",
       "      <td>H16.06.009.01.02.06.05</td>\n",
       "      <td>5</td>\n",
       "      <td>12141</td>\n",
       "      <td>MTG</td>\n",
       "      <td></td>\n",
       "      <td>aspiny</td>\n",
       "      <td>NA</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.051160</td>\n",
       "      <td>-67.468758</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-70.875002</td>\n",
       "      <td>1.891881</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.989616</td>\n",
       "      <td>-101.000000</td>\n",
       "      <td>-69.209610</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>596020931</th>\n",
       "      <td>None</td>\n",
       "      <td>[66.0, 220.0, 105.0]</td>\n",
       "      <td>Homo Sapiens</td>\n",
       "      <td>H17.06.009.11.04.02</td>\n",
       "      <td>4</td>\n",
       "      <td>12141</td>\n",
       "      <td>MTG</td>\n",
       "      <td></td>\n",
       "      <td>aspiny</td>\n",
       "      <td>NA</td>\n",
       "      <td>...</td>\n",
       "      <td>14.667340</td>\n",
       "      <td>1.336668</td>\n",
       "      <td>-63.593754</td>\n",
       "      <td>-63.239583</td>\n",
       "      <td>-75.518753</td>\n",
       "      <td>1.452890</td>\n",
       "      <td>1.441754</td>\n",
       "      <td>1.556087</td>\n",
       "      <td>-82.531250</td>\n",
       "      <td>-74.260269</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>576095991</th>\n",
       "      <td>None</td>\n",
       "      <td>[274.0, 331.0, 228.0]</td>\n",
       "      <td>Homo Sapiens</td>\n",
       "      <td>H17.06.007.11.08.01</td>\n",
       "      <td>5</td>\n",
       "      <td>12116</td>\n",
       "      <td>MFG</td>\n",
       "      <td></td>\n",
       "      <td>aspiny</td>\n",
       "      <td>NA</td>\n",
       "      <td>...</td>\n",
       "      <td>3.245000</td>\n",
       "      <td>1.053000</td>\n",
       "      <td>-59.156254</td>\n",
       "      <td>-61.437502</td>\n",
       "      <td>-67.625004</td>\n",
       "      <td>2.520616</td>\n",
       "      <td>2.586455</td>\n",
       "      <td>2.219604</td>\n",
       "      <td>-81.406250</td>\n",
       "      <td>-64.442192</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>531520637</th>\n",
       "      <td>None</td>\n",
       "      <td>[127.0, 354.0, 219.0]</td>\n",
       "      <td>Homo Sapiens</td>\n",
       "      <td>H16.06.007.01.05.03</td>\n",
       "      <td>3</td>\n",
       "      <td>12113</td>\n",
       "      <td>FroL</td>\n",
       "      <td></td>\n",
       "      <td>aspiny</td>\n",
       "      <td>NA</td>\n",
       "      <td>...</td>\n",
       "      <td>13.983713</td>\n",
       "      <td>1.427020</td>\n",
       "      <td>-62.125000</td>\n",
       "      <td>-62.718751</td>\n",
       "      <td>-77.043750</td>\n",
       "      <td>1.315267</td>\n",
       "      <td>1.307436</td>\n",
       "      <td>1.375686</td>\n",
       "      <td>-85.250008</td>\n",
       "      <td>-76.195251</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>596049209</th>\n",
       "      <td>None</td>\n",
       "      <td>[66.0, 220.0, 105.0]</td>\n",
       "      <td>Homo Sapiens</td>\n",
       "      <td>H17.06.009.11.04.05</td>\n",
       "      <td>1</td>\n",
       "      <td>12141</td>\n",
       "      <td>MTG</td>\n",
       "      <td></td>\n",
       "      <td>aspiny</td>\n",
       "      <td>NA</td>\n",
       "      <td>...</td>\n",
       "      <td>1.969320</td>\n",
       "      <td>1.048904</td>\n",
       "      <td>-59.093750</td>\n",
       "      <td>-56.583336</td>\n",
       "      <td>-62.612502</td>\n",
       "      <td>3.138165</td>\n",
       "      <td>3.238456</td>\n",
       "      <td>2.890686</td>\n",
       "      <td>-99.718758</td>\n",
       "      <td>-60.669804</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 70 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "          reporter_status     cell_soma_location       species  \\\n",
       "528642047            None    [69.0, 254.0, 96.0]  Homo Sapiens   \n",
       "596020931            None   [66.0, 220.0, 105.0]  Homo Sapiens   \n",
       "576095991            None  [274.0, 331.0, 228.0]  Homo Sapiens   \n",
       "531520637            None  [127.0, 354.0, 219.0]  Homo Sapiens   \n",
       "596049209            None   [66.0, 220.0, 105.0]  Homo Sapiens   \n",
       "\n",
       "                             name structure_layer_name  structure_area_id  \\\n",
       "528642047  H16.06.009.01.02.06.05                    5              12141   \n",
       "596020931     H17.06.009.11.04.02                    4              12141   \n",
       "576095991     H17.06.007.11.08.01                    5              12116   \n",
       "531520637     H16.06.007.01.05.03                    3              12113   \n",
       "596049209     H17.06.009.11.04.05                    1              12141   \n",
       "\n",
       "          structure_area_abbrev transgenic_line dendrite_type apical  ...  \\\n",
       "528642047                   MTG                        aspiny     NA  ...   \n",
       "596020931                   MTG                        aspiny     NA  ...   \n",
       "576095991                   MFG                        aspiny     NA  ...   \n",
       "531520637                  FroL                        aspiny     NA  ...   \n",
       "596049209                   MTG                        aspiny     NA  ...   \n",
       "\n",
       "          trough_t_ramp trough_t_short_square  trough_v_long_square  \\\n",
       "528642047           NaN              1.051160            -67.468758   \n",
       "596020931     14.667340              1.336668            -63.593754   \n",
       "576095991      3.245000              1.053000            -59.156254   \n",
       "531520637     13.983713              1.427020            -62.125000   \n",
       "596049209      1.969320              1.048904            -59.093750   \n",
       "\n",
       "          trough_v_ramp  trough_v_short_square  \\\n",
       "528642047           NaN             -70.875002   \n",
       "596020931    -63.239583             -75.518753   \n",
       "576095991    -61.437502             -67.625004   \n",
       "531520637    -62.718751             -77.043750   \n",
       "596049209    -56.583336             -62.612502   \n",
       "\n",
       "           upstroke_downstroke_ratio_long_square  \\\n",
       "528642047                               1.891881   \n",
       "596020931                               1.452890   \n",
       "576095991                               2.520616   \n",
       "531520637                               1.315267   \n",
       "596049209                               3.138165   \n",
       "\n",
       "           upstroke_downstroke_ratio_ramp  \\\n",
       "528642047                             NaN   \n",
       "596020931                        1.441754   \n",
       "576095991                        2.586455   \n",
       "531520637                        1.307436   \n",
       "596049209                        3.238456   \n",
       "\n",
       "           upstroke_downstroke_ratio_short_square  vm_for_sag      vrest  \n",
       "528642047                                1.989616 -101.000000 -69.209610  \n",
       "596020931                                1.556087  -82.531250 -74.260269  \n",
       "576095991                                2.219604  -81.406250 -64.442192  \n",
       "531520637                                1.375686  -85.250008 -76.195251  \n",
       "596049209                                2.890686  -99.718758 -60.669804  \n",
       "\n",
       "[5 rows x 70 columns]"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "human_aspiny_ephys.head(5) # Checking the dataframe to ensure the merge was successful"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "We can now begin to compare the two different neuron types by their electrophysiological data. Instead of comparing every single variable, we are choosing to compare a few key properties that reflect general basic neuron electrophysiology: resting membrane potential (vrest), input resistance (ri), membrane time constant (tau), "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Data Analysis & Results"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Include cells that describe the steps in your data analysis."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [],
   "source": [
    "## YOUR CODE HERE\n",
    "## FEEL FREE TO ADD MULTIPLE CELLS PER SECTION"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Conclusion & Discussion"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "*Fill in your discussion information here*"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.11.9"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
